esp/services/ws_workunits/ws_workunitsAuditLogs.cpp

```diff
@@ -1784,3 +1786,70 @@ bool CWsWorkunitsEx::onWUClusterJobSummaryXLS(IEspContext &context, IEspWUCluste
     }
     return true;
 }
+
+bool CWsWorkunitsEx::onWUGetThorJobQueue(IEspContext &context, IEspWUGetThorJobQueueRequest &req, IEspWUGetThorJobQueueResponse &resp)
+{
+    try
+    {
+        SecAccessFlags accessOwn, accessOthers;
+        getUserWuAccessFlags(context, accessOwn, accessOthers, true);
+
+        const char *cluster = req.getCluster();
+        const char *startDate = req.getStartDate();
+        const char *endDate = req.getEndDate();
+        unsigned maxJobQueueItemsToReturn = req.getMaxJobQueueItemsToReturn();
+        if (maxJobQueueItemsToReturn == 0)
+            maxJobQueueItemsToReturn = DefaultMaxJobQueueItemsInWUGetJobQueueResponse;
+
+        CDateTime fromTime, toTime;
+        StringBuffer fromstr, tostr;
+        if (!isEmptyString(startDate))
+            fromTime.setString(startDate, NULL, false);
+        if (isEmptyString(endDate))
+            toTime.setNow();
+        else
+            toTime.setString(endDate, NULL, false);
+
+        StringBuffer filter("ThorQueueMonitor");
+        if (!isEmptyString(cluster))
+            filter.appendf(",%s", cluster);
+
+        StringAttrArray queueInfoLines;
+        queryAuditLogs(fromTime, toTime, filter.str(), queueInfoLines);
+
+        WsWuJobQueueAuditInfo jq;
+        IArrayOf<IEspThorQueue> queues;
+        unsigned maxConnected = 0;
+        unsigned longestQueue = 0;
+        unsigned lastLineID = queueInfoLines.length() - 1;
+        ForEachItemIn(idx, queueInfoLines)
+        {
+            const char* line = queueInfoLines.item(idx).text;
+            if(isEmptyString(line))
+                continue;
+
+            jq.getAuditLineInfo(line, longestQueue, maxConnected, maxJobQueueItemsToReturn,
+                idx < lastLineID ? 1 : 2, queues);
+
+            if (queues.length() > maxJobQueueItemsToReturn)
+                break;
+        }
+
+        if (queues.length() > maxJobQueueItemsToReturn)
```

Comment:
Could this code be moved to the conditional in line 1834?

Reply:
Will do.